src/javascripts/components/mushroomList/mushroomList.js

```diff
@@ -50,17 +53,23 @@ const showShroomForm = (e) => {
 const addShroomEvent = (e) => {
   e.preventDefault();
 
+  const file = document.getElementById('mush-image').files[0];
+  const image = file.name;
+
   const newMush = {
     name: $('#mush-name').val(),
     size: $('#mush-size').val(),
     location: $('#mush-location').val(),
     weight: $('#mush-weight').val() * 1,
+    image,
   };
 
   mushroomData.addMushroom(newMush)
     .then(() => {
-      // eslint-disable-next-line no-use-before-define
-      buildForest();
+      firebase.storage().ref(`mushrooms/${image}`).put(file).then(() => {
```

Comment:
this is the magic/where the upload happens. We create a new reference to where we want firebase storage to save the image with ```firebase.storage().ref(`mushrooms/${image}`)```, then we `put` the file to that location. This returns a promise, so once that promise is successful we rebuild the forest.